tests/command_line/check_main.py

```diff
@@ -101,6 +102,12 @@ def check_immutable(self):
         with pytest.raises(NotImplementedError):
             global_dict["y"] = 3
 
+    def check_pickleable(self):
+        """Expect the user defined dict object to be pickleable"""
+        globals_dict = get_user_defined_globals(globals_json)
+
+        assert pickle.loads(pickle.dumps(globals_dict)) == globals_dict
```

Comment:
In order to be pedantic about this test, it might be useful to also `assert globals_dict`. Even in py3 pickling works ok with pysistence if the dictionary is empty.